FWCore/ParameterSet/interface/DocFormatHelper.h

```diff
@@ -82,7 +83,11 @@ namespace edm {
     static int offsetTopLevelPSet() { return 2; }
     static int offsetSectionContent() { return 4; }
 
+    void addCategory(std::string const& pluginCategory);
+    bool wasCategoryAlreadyPrinted(std::string const& pluginCategory) const;
+
   private:
+    std::vector<std::string> pluginCategoriesAlreadyPrinted_;
```

Comment:
Would it make sense to use `std::unordered_set`? _Maybe_ we won't see plugins with so many helper plugins that it would make a difference, but the linear search on an unordered `std::vector` made we wonder this out loud.

Reply:
It is hard to predict the future, but currently we have 4 modules using 1 plugin category and 1 module using 2 plugin categories. So the size of the container is 0 more than 99% of the time and rarely 1 or 2. I expect the number of modules using them to increase over time, but I'd be surprised to see many plugin categories in the same module. There are only 6 categories possible outside Core tests currently (max size of container could be 6).

A linear search in a vector that small might be the most efficient. And I think a vector uses the least resources (memory). I don't know how the hash tables work but I imagine the overhead to set up the table is more than for a vector. The most significant criteria may be how much resources does it use when it is empty.

I am almost done implementing Chris's suggestion and the container will become a vector<pair<string,string>> in a few minutes.